src/main.js

```diff
@@ -74,15 +76,31 @@ const createWindow = async () => {
       ],
     },
   });
-
   const menubar = Menu.buildFromTemplate(
     menuTemplate(mainWindow, ELECTRON_DEV_MODE)
   );
   Menu.setApplicationMenu(menubar);
 
-  // and load the index.html of the app.
+  splashScreen = new BrowserWindow({
+    width: 574, // dims set to match the image in splash.html
+    height: 500,
+    transparent: true,
+    frame: false,
+    alwaysOnTop: true,
+  });
+  splashScreen.loadURL(`file://${__dirname}/static/splash.html`);
   mainWindow.loadURL(`file://${__dirname}/index.html`);
 
+  mainWindow.once('ready-to-show', () => {
+    splashScreen.destroy();
+    // We should be able to hide mainWindow until it's ready,
+    // but there's a bug where a window initialized with { show: false }
+    // will load with invisible elements until it's touched/resized, etc.
+    // https://github.com/electron/electron/issues/27353
+    // So for now, we have the splashScreen over a blank, white mainWindow.
+    // mainWindow.show();
```

Comment:
should this still be commented out?

Reply:
Yes, but I updated the comment so that's no longer confusing.